### PR TITLE
Feature/conductor execution api v1alpha2 integration

### PR DESCRIPTION
**Pull request by steezeburger**
## Summary
This PR is the integration PR for Execution API v1alpha2. The logic has been reviewed and approved, but there may be some Rust improvements to be made.

## Background
v1alpha1 was not sufficient for our needs, so v1alpha2 was created

## Changes
- removes v1alpha1 api
- adds v1alpha2 api
- implements functionality to use v1alpha2

## Testing
- fixes existing tests
- added new test for scenario where we get an already executed block from da

## Breaking Changelist
- v1alpha1 api methods removed

## Related Issues
https://github.com/astriaorg/astria/issues/205
https://github.com/astriaorg/astria/issues/207

closes https://github.com/astriaorg/astria/issues/205
closes https://github.com/astriaorg/astria/issues/207

### CODE REVIEW

File: crates/astria-conductor/src/execution_client.rs
Comment (noot) [inline crates/astria-conductor/src/execution_client.rs:19]: since this trait is `pub(crate)` you can remove the `Sealed` trait

Comment (SuperFluffy) [reply crates/astria-conductor/src/execution_client.rs:19]: @noot this is already done in the single-entry-point PR

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:164]:
```suggestion
                    match self.execute_block(block_subset).await {
```

why are we cloning now? looks unnecessary

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:171]:
```suggestion
                            .await.wrap_err("failed to update commitment state")?;
```

Comment (noot) [reply crates/astria-conductor/src/executor.rs:171]: actually, i think returning an error here stops the whole event loop and thus the entire conductor - is that what we want, or should we just log here? the error below is just logged, so maybe we do that?

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:235]: can you just clone the hash instead of the block?

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:241]: no `unwrap` pls

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:281]:
```suggestion
       self.commitment_state = self
            .execution_rpc_client
            .call_update_commitment_state(CommitmentState {
                firm: Some(firm),
                soft: Some(soft),
            })
            .await?
```

nit

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:302]:
```suggestion
                // remove the sequencer block hash from the map, as it's been firmly committed
```

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:327]: this looks weird - i also noticed we are usually only updating `soft` OR `firm`, not both at the same time (except for here), so i'd add functions `update_soft_commitment` and `update_firm_commitment` to make it clear which is being updated

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:1]: High level comment on this module: right now this is basically defining a newtype wrapper around `ExecutionServiceClient` + a trait `ExecutionClient` that is then impl'd on the wrapper. The wrapper itself carries no further state.

I think this PR touches so many bits that it's justified to either:

1. impl the trait methods as inherent methods on the wrapper (without th trait)
2. or impl `ExecutionClient` as an extension trait on `ExecutionServiceClient` (without th wrapper)

File: crates/astria-conductor/src/executor.rs
Comment (SuperFluffy) [reply crates/astria-conductor/src/executor.rs:235]: I think cloning in general is unnecessary. tendermint's `Hash` impls `AsRef<[u8]>` and `hex::encode` is defined for `T: AsRef<[u8]>`

Comment (steezeburger) [reply crates/astria-conductor/src/executor.rs:230]: `execution_block.hash` is a `Vec<u8>` here

Comment (SuperFluffy) [reply crates/astria-conductor/src/executor.rs]: Doesn't change the argument tho: https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-AsRef%3C%5BT%5D%3E-for-Vec%3CT,+A%3E :-)

File: crates/astria-conductor/src/executor.rs
Comment (steezeburger) [reply crates/astria-conductor/src/executor.rs]: updated! to avoid nesting, i refactored the match into some `if lets`. please lmk if this is better @noot

File: crates/astria-conductor/src/execution_client.rs
Comment (steezeburger) [reply crates/astria-conductor/src/execution_client.rs:1]: updated! i went with option 2. lmk if this is what you intended! @SuperFluffy

**Comment by joroshiba** [comment]
High level LGTM, reviewed all the previous PRs into this feature branch.

I think we should block merging this on ensuring it works with [astriaorg/go-ethereum#](https://github.com/astriaorg/go-ethereum/pull/29). Hopefully this "just works", but we should do some testing to make sure. May require changes to both. 

I want to avoid ending up in a state where where we can't cut a new conductor release because there isn't a geth image that works with it (especially with other changes currently going in).

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:247]: is there ever a case this could happen? seems concerning if there is, as nothing can get executed

File: crates/astria-conductor/src/executor.rs
Comment (noot) [inline crates/astria-conductor/src/executor.rs:284]: wrap_err

File: crates/astria-conductor/src/executor.rs
Comment (steezeburger) [reply crates/astria-conductor/src/executor.rs:247]: I don't think there is a case it would ever occur. That type comes from the proto definitions. I think I misunderstood your comment and added this complex handling instead of just using `wrap_err` haha. Should I just use `wrap_err` here?

**Review by noot — APPROVED** [review]
looks good! like @joroshiba said we should probably test w/ geth before merging

File: crates/astria-conductor/src/executor.rs
Comment (noot) [reply crates/astria-conductor/src/executor.rs:247]: maybe return an error here - looking at the code, it seems this case shouldn't ever occur, so would be developer error if it does

Comment (steezeburger) [reply crates/astria-conductor/src/executor.rs:247]: It should not ever happen no. What's a better way I can handle the `Option` here?

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:23]: Small nit: I think I would personally just use `ExecutionServiceClient::connect` at the callsite and forego this extra wrapper.

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:30]: Small nit: might call this `ExecutionClientExt` to make it clear this is an extension trait, but no big deal. Many other crates in the ecosystem would also just call this `Client` without `Ext`.

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:67]: error context, i.e. `await.wrap_err(...)`

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:89]: error context

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:102]: error context :)

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:109]: error context

File: crates/astria-conductor/src/execution_client.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/execution_client.rs:125]: you guessed it: error context :P

File: crates/astria-conductor/src/executor.rs
Comment (SuperFluffy) [inline crates/astria-conductor/src/executor.rs:129]: error context

File: crates/astria-conductor/src/executor.rs
Comment (SuperFluffy) [reply crates/astria-conductor/src/executor.rs:247]: It looks like executor is handling too many of the protobuf idiosyncracies like `soft` and `hard` being `Option`. If it basically never makes sense for these fields to be unset, then it's better to have one extra step that converts from the `protobuf-CommitmentState` to an `idiotmatic-CommitmentState` (failing if one of the fields is not set).

This way you don't have to worry about all of that `None` business.

**Review by sambukowski — APPROVED** [review]

**Comment by joroshiba** [comment]
#528 merged instead, contained conflict updates.

**Pull request closed without merging**
